src/System.Data.Common/src/System/Data/SQLTypes/SQLDecimal.cs

```diff
@@ -2267,7 +2267,15 @@ internal void AdjustScale(int digits, bool fRound)
                 throw new OverflowException(SQLResource.ArithOverflowMessage);
 
             bNewScale = (byte)(lAdjust + _bScale);
-            bNewPrec = (byte)(Math.Min(s_NUMERIC_MAX_PRECISION, Math.Max(1, lAdjust + _bPrec)));
+
+            if (Value == Decimal.Zero && _bScale == 0 && lAdjust > 0)
+            {
+                bNewPrec = bNewScale;
```

Comment:
This will also create SqlDecimal with precision 2 & scale 0.
```
SqlDecimal value = SqlDecimal.ConvertToPrecScale(SqlDecimal.Parse("0"), 2, 0);
```
I will rollback the code change to the initial version for those reasons:
* When `SqlDecimal` 0 with precision 2 & scale 0 is being inserted to a table with `decimal(2,2)` type, it should be inserted successfully by converting the precision & scale to 2 & 2. If I add the condition `_bPrec == 1` to the if statement, this scenario will be failed since it will be converted to 3 & 2 instead of 2 & 2 by following previous logic.
* We cannot use decremented precision by 1 in the plain zero scenario, since the initial precision can be larger than 1.
* We do not need to know what the initial precision of plain zero with scale 0 is, when the scale is being adjusted by positive amount, because the new precision should have ***always the same value with the scale*** in this case.
* We do not need to waste system resource by calling `Math.Min()` and `Math.Max()` methods in plain zero scenario, because the new precision will have the same value of the new scale, and the new scale is always larger than 0 and less or equal to `s_NUMERIC_MAX_PRECISION` when the scale is being adjusted by positive amount.